packages/aws-cdk/lib/cli/io-host/cli-io-host.ts

```diff
@@ -154,6 +168,45 @@ export class CliIoHost implements IIoHost {
     this.stackProgress = props.stackProgress ?? StackActivityProgress.BAR;
   }
 
+  /**
+   * Required for telemetry
+   */
+  public async bindTelemetryClient(argv: any, context: {[key: string]: any}) {
+    // TODO: change this to EndpointTelemetryClient
+    this.telemetryClient = new IoHostTelemetryClient({
+      ioHost: this,
+    });
+
+    // sanitize the raw cli input
+    const command = sanitizeCommandLineArguments(argv, await makeConfig());
+    this.telemetryInfo = {
+      identifiers: {
+        installationId: getInstallationId(this.asIoHelper()),
+        sessionId: randomUUID(),
+        telemetryVersion: '1.0',
+        cdkCliVersion: version.versionNumber(),
+        accountId: await new AccountIdFetcher().fetch(),
+        region: await new RegionFetcher().fetch(),
+      },
+      event: {
+        command: {
+          path: command.path,
+          parameters: command.parameters,
+          config: sanitizeContext(context),
+        },
+      },
+      environment: {
+        ci: Boolean(process.env.CI),
```

Comment:
`isCI();` and double check that the codebuild / github wf / codebuild within pipeline have this flag

Reply:
look at other tools that do ci detection

Reply:
ciSystem

Reply:
i think what i have now is best i can do without doing any actual work :)